vidaud/seta
# The initial stage of refactoring for the seta-react project

## Seta React

### Description

This PR introduces the first stage of proposed refactoring to the React project.

The following changes have been performed:

- Introduced linting and formatting rules to verify the code quality and consolidate the coding style - please see the `/docs/eslint-configuration.md` file for instructions
- Updated most of the codebase to adhere to the linting and formatting rules
- Implemented a pre-commit hook to prevent commiting code that fails the rules validation
- Fixed hot-reloading for the client app, so that code changes are instantly available in the browser during development without reloading the page (and losing the state)
- Fixed the `action` parameter remaining in the URL after login
- Changed all the components/pages to adhere to the general React conventions for file-naming and structure
- Removed dead code
- Introduced a couple of helper scripts to `package.json` to easily build/start/stop the dev containers from the React project's folder

### How to test

- Make sure the app runs normally and the general functionality behaves as expected - almost all the files have been touched by fixing the linting problems and reorganizing the components, but most of these are not functional changes and should not impact the app's behavior
- Make sure the `?action=login` parameter no longer remains in the browser's URL bar after login
- Verify that changing a file in the project reflects immediately in the broswer - eg. adding a `div` with some text on a page
- Verify that executing `npm run docker:up` from the folder `seta-react` works as expected (also, `docker:build` and `docker:down`)
